internal/orchestrator/commiter.go

```diff
@@ -42,13 +45,203 @@ func (c *Commiter) Start() {
 	go func() {
 		for t := range ticker.C {
 			fmt.Println("Commiter running at", t)
-			// TODO: fetch max block number from main table
-			// TODO: fetch sequential block numbers from staging table
-			// TODO: save to main table
-			// TODO: delete from staging table
+			blocksToCommit, err := c.getSequentialBlocksToCommit()
+			if err != nil {
+				log.Printf("Error getting blocks to commit: %v", err)
+				continue
+			}
+			if len(blocksToCommit) == 0 {
+				log.Println("No blocks to commit")
+				continue
+			}
+			if err := c.commit(blocksToCommit); err != nil {
+				log.Printf("Error committing blocks: %v", err)
+			}
 		}
 	}()
 
 	// Keep the program running (otherwise it will exit)
 	select {}
 }
+
+func (c *Commiter) getBlockNumbersToCommit() ([]uint64, error) {
+	maxBlockNumber, err := c.storage.DBMainStorage.GetMaxBlockNumber()
+	if err != nil {
+		return nil, err
+	}
+	startBlock := maxBlockNumber + 1
+	endBlock := maxBlockNumber + uint64(c.blocksPerCommit)
```

Comment:
In the design document you had the conversation with jake to do backfills from the oldest block not the end to follow the chain behaviour

Reply:
It gets the max block from the main table to determine where should it start commiting from.
Lets say last cycle it commited up to 1000. Then next cycle it should try to commit from 1001